### ISSUE TITLE
drop Catalyst 

### ISSUE THREAD

**Issue by philmmanjaro** [manjaro/release-plan#37]
Since AMD is now focusing on amdgpu we might drop support for the depreciated Catalyst driver. Let's find a way to support amdgpu-pro instead

**Comment by udeved** [comment]
Related question
https://github.com/manjaro/release-plan/issues/39

**Comment by Kirek** [comment]
We should drop Catalyst as soon as they release an amdgpu pro driver with support for [Polaris](https://en.wikipedia.org/wiki/AMD_Radeon_Rx_400_Series) and/or something labelled stable by them.
Or we go with the recently released beta version?

**Comment by udeved** [comment]
Probably a matter of opinion.
I wouldn't go with with beta drivers, should there be any ugly issue with it, eg overheating, people will blame you for deploying beta driver despite beta.

I updated yesterday from arch testing xorg but not nvidia.
Latest stable Nvidia driver still produces the same strange results as with beta drivers.

**Comment by philmmanjaro** [comment]
So what ever we do newer drivers create issues. I thought Nvidia knows about the regression already. Did we already report it?

**Comment by udeved** [comment]
https://github.com/manjaro/release-plan/issues/142

**Comment by mirh** [comment]
Amdgpu-pro transition is being handled in https://github.com/manjaro/packages-community/pull/276
And as we all know, it will never come to make a difference for older gpus (which at least for a couple of other months -in airlied we trust- can even be stuck to GL 3.3 on *radeon*)

~~We~~ Phil managed to avoid libgvnd catastrophe, and by now the X.org replacement system is well in place, thus for the foreseeable future I don't see breakages in sight. Kernels aside - and in Vi0L0 we trust (but since afaik 17.1 isn't going to ship with anything newer than 4.12 anyway, catalyst should work there too)

So.. I'm not sure why talking about this now?
Moreover, if the driver will end to magically still stick together by ~2018, I believe we might even see it _improves_, thanks to the ongoing work of acpica guys.

**Comment by Tids** [comment]
For catalyst 
https://github.com/Tids/packages-extra/commit/f3a98c55347c4008849b5b78361bd5dda6c1e593
https://github.com/manjaro/packages-extra/compare/master...Tids:master

This removes all PCI-IDs from the catalyst.ids files that is supported by radeonsi (amdgpu/+exp), but keeps everything thats older. So that GCN generations are supported only by radeon/amdgpu where they get the best performance and new users are not confused by the catalyst driver anymore.

Feedback would be nice about this idea.

**Comment by mirh** [comment]
I'll confess it took me 5 good minutes to realize wtf catalyst-utils had to do with mhwd. 
Guess it was at least quick to implement. 

Said this, cool very nice idea. Should have been thought earlier. 
~~But with ~~manjaro~~ Arch still not supporting [AMDGPU-PRO](https://github.com/corngood/archlinux-amdgpu) (and ROCm only being GCN 3rd gen and newer), I still see why one could still desire fgrlx to do some opencl work.~~
Ok, I see there's a [standalone package](https://aur.archlinux.org/packages/opencl-amd/) for that. 

And even then, AFAIK first two GCN gen are still very half-assed with amdgpu driver. 
...

Ok TL;DR, maybe we should wait either for that to stabilize, and/or for DC to land in mainline (hopefully in a month or so)?

**Comment by Tids** [comment]
> I'll confess it took me 5 good minutes to realize wtf catalyst-utils had to do with mhwd.
Its in this case just the provider for the PCI-IDs, the list that tells mhwd for what hardware we show what config.

AMDGPU-Pro is made for Enterprise distros only. It will never fully work on something other than this. Its made for people who pay to get support for the drivers, so Manjaro is and never will be a target for them and I also dont plan to add support for it, since it is useless and gain, a holdback, for us.

DisplayCode will come with 4.15 (hopefully) so the main part for amdgpu-pro is finally here ins early 2018.
ROCm needs a bit more work (on llvm as well as the linux side, but this will happen in 2018
fglrx OpenGL is something we dont even want
AMD Vulkan would be interesting, but since valve is working on RadV, I'm not very interested in this

So after all I dont see the reason in amdgpu-pro for manjaro.


for amdgpu-experimental support; i think they will leave it as it is and thats not bad at all, radeon works great for GCN1.1,1.0 and radeonsi is still much faster than catalyst-fglrx-opengl.

The OpenCL part is something, but its a really special part, from what i see in the forums OpenCL is almost meaningless compared to OpenGL for Manjaro Users and thats my point.

**Comment by mirh** [comment]
> Its in this case just the provider for the PCI-IDs, the list that tells mhwd for what hardware we show what config.

Yes, I had got that then. 
~~Just.. Relying on that feels very kind of odd, given it's not like _somebody_ is gonna add ***ever*** any more IDs?~~
And indeed you are switching to PCI IDs.. 
My total hats off*

> AMDGPU-Pro is made for Enterprise distros only. 

Tbh, I always thought to it as the continuation of fglrx.. 
Though maybe you are right. Once DC lands.. Sure, maybe it'll have some of tiniest fine tuned optimizations, but really, no big deal at all (contrarily to the past). 
Funnily then, only 3 hours after my post, a guy on the repo I linked claimed to have finally set up everything (working with both 4.13 and x.org 1.19)
But whatever, sure. 

>  fglrx OpenGL is something we dont even want

I puke to the thought. 

> The OpenCL part is something, but its a really special part, from what i see in the forums OpenCL is almost meaningless compared to OpenGL for Manjaro Users and thats my point.

Mhh, _yeah_ I see perhaps I might be kind of overthinking here. 
With 2 users reporting CL problems likely being exactly the whole user-base, and the remaining silent majority that just care for GL living happily. 
Guess like we could already proceed then. 

But are you sure SI is.. not saying necessarily really _production ready_.. But at least usably stable?

---
*except we have a big problem, with fglrx only supporting DID whitelisted in `\common\etc\ati\control`
Now, for the love of me it's half a week I'm trying to figure out which kind of encoding they used in that file, but nevertheless long story short, it means catalyst should be "opt-in", not "opt-out". 

p.s. it's incredible how many gpus amd official Setup Information files are lacking.

**Comment by philmmanjaro** [comment]
@Tids: should we drop it after all with **Illyria** series?

**Comment by Tids** [comment]
Depends... While I would always say "yes", it would be maybe better to wait for the FP64 emulation in Mesas R600G > https://www.phoronix.com/scan.php?page=news_item&px=Airlie-Preps-Soft-FP64-Mesa

So new User will drop to OpenGL4.3 instead of 3.x without Catalyst

**Comment by mirh** [comment]
You'd jump to 4.4 actually (even though fp64 is so useless in itself, that in all fairness you could just as well already fake report it)
Still, we are missing out openCL. And contrarily to some months ago (I don't want to sound bossy) I found myself actually using it. 

Also, as I was saying above (god, I'm really verbose), uninstall is still bugged https://github.com/manjaro/mhwd/issues/28

**Comment by Tids** [comment]
OpenCL is something we will never get on radeon. But do you think it's smart to ship catalyst just because of this? I mean it'll now be the 3rd year of no updates for it.

If we would put it to the AUR - so it would be reachable for everyone that is interested - would that change much?

**Comment by mirh** [comment]
> OpenCL is something we will never get on radeon.

TBH they are still working on it. Slow AF, but still something. I had posted something on the forums the other day. 
(and then, clover itself is shared by all gallium drivers - including \*every\* nvidia cards)

> But do you think it's smart to ship catalyst just because of this? I mean it'll now be the 3rd year of no updates for it.

Well, it still works. I totally acknowledge your bad experiences then, but I guess like there are use cases and use cases. 
If any the question is if it's still "worth the effort", just for that. 
And I see, phil is pretty.. Proactive and present fixing it, so maybe if he's asking about the drop he's full of it (or perhaps he's just not used to keep updated on mesa-dev :c)

*Though* even with just Vi0L0 keeping the boat afloat.. I **genuinely** cannot see how hard it could be to do a couple of copypastas?

> If we would put it to the AUR - so it would be reachable for everyone that is interested - would that change much?

Perhaps. It would be "refreshing" to put it outside manjaro repos. 
But the thing is, I'm kinda sure arch would not accept anything crafted for only a distro (which would be definitively required, not to piss off mhwd)

**Comment by mirh** [comment]
> Just as they work on GNU-Hurd ... But AMD is not working on it, its years behind and shared code is worth nothing as long as nobody can use it and for clover, i have no hope that it will be usable in my lifetime..

Just this month, there have been fifty [unique] commits on the mailing list. 

> I mean we could just go back to xorg 1.5, stay on this one for ever.

*confess to have tried to do exactly [that](https://github.com/manjaro/release-plan/files/1864033/linux316-extramodules.zip)*
Too bad the guy in the [thread](https://forum.manjaro.org/t/ati-graphics-card-drivers-how-to/32784) was just that stupid. 

> wich is already the case - just look at bumblebee, very old nvidia drivers, catalyst, x86 - and just to say that: you dont want to be the plattform where people with old hardware switch to.

I haven't any particular experience with any of the other, and to be honest until just now I thought they had no particular problematic (hadn't bumblebee been replaced by prime maybe?)
And not sure what's the problem with users with old fashioned hardware, as long as driver is open. 

On the other hand, after quite a thinking* I think I got the idea/lesson/point. So.. I think maybe everybody could be happy already with a deadline?
What about keeping it up to kernel 4.14**?
When you are on anything higher, mhwd will hide it. 

And then, anyway, if it really is that easy to support, I guess like it wouldn't be too hard for me to make a PR ***/s***

*(and noticing that starting with 4.15 a `BUG: using smp_processor_id() in preemptible code: Xorg/535` / `caller is kas_spin_lock[fglrx]` error appears, for as much as functionally innocuous)
**very coincidentally an LTS release with a support cycle ending the same month than \*holy windows 7\* one

**Comment by Strit** [comment]
It's not like the nvidia drivers are going away soon.

Nvidia just won't provide updates for the 32-bit versions from 2019, so it's up to the 32-bit distro's to decide if they want to maintain a possibly "broken" package.
If they don't, then yeah, the ability to quickly switch to open source drivers are still there.

**Comment by mirh** [comment]
They aren't going away.. But the remainder software is?
Bringing you to the same exact situation of fglrx. 

And it's [not](https://nvidia.custhelp.com/app/answers/detail/a_id/3142) [only](http://nvidia.custhelp.com/app/answers/detail/a_id/4654) about x86 as I was saying?

**Comment by Tids** [comment]
What's the status support for this new legacy hardware in noveau? I mean for real, if things are not supported set up a hint that this hardware is not supported and don't choose to make them run in a frustrating way.
New users who are pissed about Manjaro because it claims support and then the result is like with fglrx are no users you want to have.

**Comment by mirh** [comment]
I'm subscribed to all catalyst tags on the forums, and tbh I have never seen "angered" people. 
(and as I already said somewhere, I'm pretty satisfied with it, for as much as my use case is a bit less consumer than you'd expect)

Anyway, I really wouldn't know. For fermi I'm sure nouveau sucks. 
But for Tesla and older, I hardly got to hear of them in latest years. According to [this random snippet](https://trello.com/c/A4lqI9kt/89-tesla-1st-gen-memory-reclocking) memory reclocking is only working on GDDR3 cards. 
Hardly good. 
Status unknown for even older cards (Curie), but we have right now a dude on the forums going to report about them.
